applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -234,16 +230,16 @@ static void base_recv_cb(struct bt_audio_broadcast_sink *sink, const struct bt_a
 			LOG_DBG("BIS %d   index = %d", j, index);
 
 			/* If this is a BIS of interest then attach to and start a stream */
-			if (index == channel) {
+			if (index == channel_bit) {
 				if (bitrate_check((struct bt_codec *)&base->subgroups[i].codec)) {
 					base_bis_index_bitfield |= BIT(index);
 
-					streams[i].codec =
+					audio_stream.codec =
 						(struct bt_codec *)&base->subgroups[i].codec;
-					print_codec(streams[i].codec);
+					print_codec(audio_stream.codec);
 
-					LOG_DBG("Stream %d in subgroup %d from broadcast sink", i,
-						j);
+					LOG_DBG("Channel %s in subgroup %d from broadcast sink",
+						((channel == AUDIO_CH_L) ? CH_L_TAG : CH_R_TAG), i);
```

Comment:
Maybe we should break out of the for-loops here, so that we are sure we only sync to one stream?